src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.Android.cs

```diff
@@ -43,11 +43,6 @@ partial void UpdatePlatform()
 			var newState = new BorderLayerState(drawArea.Size, _borderInfoProvider);
 			var previousLayoutState = _currentState;
 
-			if (newState.Equals(previousLayoutState))
-			{
-				return;
-			}
-
```

Comment:
This is concerning.

Reply:
Not really. The original code is more surprising actually. Just because the "layout slot" didn't change, doesn't mean that what is being drawn is the same. In fact, there is no sound way to make this sort of optimization in the BorderLayerRenderer. If an ImageSource fires `Invalidated`, then it known better. The BorderLayerRenderer (which has less information) cannot decide that the invalidation is unnecessary.

Reply:
In that case, BorderLayerRenderer should subscribe to `ImageSource.Invalidated` event, and set a flag `forceUpdate`. Then we do `if (forceUpdate || newState.Equals(previousLayoutState))`. That way, we fix that bug and still maintain the perf optimization (even if it's quite ugly to do so)

Reply:
That's almost already the case (`Update` is only called when it really needs updating) except for this:
```csharp
#if UNO_HAS_ENHANCED_LIFECYCLE
		_owner.SizeChanged += (_, _) => Update();
#else
		// Using SizeChanged on other platforms SHOULD work. But it didn't work on Android
		// for unknown reason. For now, we are using SizeChanged only on enhanced lifecycle
		// platforms where we are sure it works correctly.
		_owner.LayoutUpdated += (_, _) => Update();
#endif
```

So I'm going to add a `forceUpdate` flag, but it would be a lot cleaner to figure out why SizeChanged doesn't work and to just call `Update` properly.

Reply:
Actually, upon further reflection, we would be inserting `forceUpdate` in almost all calls and it would be very error-prone to accurately track which calls should `forceUpdate`. I guess the better way of doing that would be making the ` newState.Equals(previousLayoutState)` check only in the LayoutUpdated subscription.

Reply:
I'm not following why subscribing to ImageSource.Invalidated won't work

Reply:
I think when padding changes and size remains the same, that should mean the invalidation should not be needed - as only the size of the content adjusted, but things like background/border still remain the same with the new padding. While it would be the best to `Update` only when actually needed, there seems to be just too many options that can trigger it, so the layer comparison + explicit invalidation based on brush changes should be sufficient (as long as all the bases are covered). `Brushes` are the only ones "problematic" enough, as they can change while the instance remains the same, but otherwise the layer state properties can be compared quite reliably

Reply:
I have made the requested changed. For the record, continuing to do these unsound ad-hoc optimizations without even profiling will continue to bite us.